sources/imperative/reader/ReaderAPI.swift

```diff
@@ -71,6 +71,46 @@ extension CSVReader {
             try self.init(configuration: configuration, buffer: buffer, decoder: decoder)
         }
     }
+
+    public enum Stream {
+        case stdin
+    }
+
+    /// Creates a reader instance that will be used to parse the given CSV stream.
+    ///
+    /// If the configuration's encoding hasn't been set and the input data doesn't contain a Byte Order Marker (BOM), UTF8 is presumed.
+    /// - parameter input: The stream type to be parsed.
+    /// - parameter configuration: Recipe detailing how to parse the CSV data (i.e. encoding, delimiters, etc.).
+    /// - throws: `CSVError<CSVReader>` exclusively.
+    public convenience init(input: Stream, configuration: Configuration = .init()) throws {
+        if configuration.presample {
+            // A. If the `presample` configuration has been set, read all data from STDIN.
+            try self.init(input: FileHandle.standardInput.readDataToEndOfFile(), configuration: configuration); return
+        } else {
+            // B. Otherwise, create an input stream and start parsing byte-by-byte.
+            guard let stream = InputStream(fileAtPath: "/dev/stdin") else { throw Error._invalidStdin() }
```

Comment:
Unsure if this should be a throwing error or a `fatalError` if stdin isn't available. I guess I'm not sure how it's reachable. I think even a closed stdin will still be able to construct a stream for. It will just be `streamStatus == .closed`.

Reply:
I prefer to throw errors if the library user can recover from it and do something instead. For the specific situation of `stdin`: what is the `streamStatus` right after you create it? Do you have to open it?

If we decide to expose `InputStream` we could have a check and let pass anything that is not `.error`, or `.closed`, or `.atEnd`.

Reply:
Looks like it's starts not open, same as a regular file's input stream.

```swift
let stream = InputStream(fileAtPath: "/dev/stdin")!
stream.streamStatus // .notOpen
stream.open()
stream.streamStatus // .open
```

> If we decide to expose InputStream we could have a check and let pass anything that is not .error, or .closed, or .atEnd.

Oh, I thought you wanted to avoid that. Happy to have either.